mh/constants.py

```diff
@@ -164,13 +166,43 @@ def make_binary_npc_greeters(is_jap=False):
     US_OFFSET = 0x180
     JP_OFFSET = 0x100
     offset = JP_OFFSET if is_jap else US_OFFSET
+
+    day_in_cycle = int(current_server_time()//SECONDS_PER_DAY) % JHEN_EVENT_OFFSET
+    is_jhen_active = day_in_cycle >= JHEN_START and day_in_cycle < JHEN_END
```

Comment:
You should create a helper method for detecting when Jhen is active

Reply:
Done